rest/src/main/kotlin/request/RequestHandler.kt

```diff
@@ -16,4 +22,11 @@ public interface RequestHandler {
      */
     @Throws(RestRequestException::class)
     public suspend fun <B : Any, R> handle(request: Request<B, R>): R
+    public suspend fun <T> intercept(builder: RequestBuilder<T>): RequestBuilder<T> = builder.apply {
```

Comment:
Does it need to be `suspend`?
Should it really return a `RequestBuilder` or just take it as an argument?

Reply:
someone may need to do some suspending logic; or so I thought

Reply:
But what about the returned builder, is that needed?

Reply:
technically no since you are expected to edit the builder in place